### PR TITLE
`Modal`, `Flyout`: make title an H1

### PR DESCRIPTION
**Pull request by shleewhite**
### :pushpin: Summary

If merged, this PR would change the HTML element wrapping the Modal and Flyout tagline + title in the heading to an H1 (the previous element was a div).

### :hammer_and_wrench: Detailed description

* update the Modal and Flyout title to be H1s
* add tests to ensure the correct tag is rendered
* update the Flyout showcase page heading structure so it doesn't jump from H1 to H3 (this causes an automated a11y test failure)

### :camera_flash: Screenshots

**New Modal markup:**
Screenshot 2024-08-22 at 12 10 24 PM

### :link: External links

RFC: https://go.hashi.co/rfc/ds-087
Jira ticket: [HDS-3773](https://hashicorp.atlassian.net/browse/HDS-3773)

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3773]: https://hashicorp.atlassian.net/browse/HDS-3773?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by shleewhite** [comment]
The Percy change I approved was a slight spacing change that happened when I update the heading structure of the Flyout showcase page. There were no changes to the actual component styles (the parts of the page that changed were all after the first heading I changed).

File: showcase/tests/integration/components/hds/flyout/index-test.js [hunk 1]
```diff
@@ -113,6 +113,15 @@ module('Integration | Component | hds/flyout/index', function (hooks) {
     assert.dom('.hds-flyout__tagline').doesNotExist();
   });
 
+  test('it renders the title as an h1', async function (assert) {
```
Comment (didoo) [inline showcase/tests/integration/components/hds/flyout/index-test.js:116]: [suggestion]
```suggestion
  test('it renders the title as an h1 by default', async function (assert) {
```

File: showcase/tests/integration/components/hds/modal/index-test.js [hunk 2]
```diff
@@ -100,6 +100,15 @@ module('Integration | Component | hds/modal/index', function (hooks) {
     assert.dom('.hds-modal__tagline').hasText('Tagline');
   });
 
+  test('it renders the title as an h1', async function (assert) {
```
Comment (didoo) [inline showcase/tests/integration/components/hds/modal/index-test.js:103]: [suggestion]
```suggestion
  test('it renders the title as an h1 by default', async function (assert) {
```

File: packages/components/src/components/hds/alert/title.hbs [hunk 3]
```diff
@@ -2,8 +2,8 @@
   Copyright (c) HashiCorp, Inc.
   SPDX-License-Identifier: MPL-2.0
 }}
-{{! IMPORTANT: we removed any extra newlines before/after the `let` to reduce the issues with unexpected whitespaces (see https://github.com/hashicorp/design-system/pull/1652) }}
-{{#let (element this.componentTag) as |Tag|}}<Tag
-    class="hds-alert__title hds-typography-body-200 hds-font-weight-semibold"
-    ...attributes
-  >{{yield}}</Tag>{{/let}}
\ No newline at end of file
+{{~#let (element this.componentTag) as |Tag|}}
```
Comment (didoo) [inline packages/components/src/components/hds/alert/title.hbs:5]: [suggestion] I know @KristinLBradley suggested to use the `~` operator, and I know this is much more readable and "nice", but personally in this specific case I would revert this change: Ember has a weird way of handling whitespace, and the one you're using doesn't assure 100% that will remove it (see how it's used [in other parts of the codebase](https://github.com/search?q=%7E+user%3Ahashicorp+repo%3Ahashicorp%2Fdesign-system+language%3AHandlebars&type=code&ref=advsearch&l=Handlebars))

if y

**Review by didoo — APPROVED** [review]
Left a few minor non-blocking suggestions. Apart from that, LGTM 👍

File: showcase/tests/integration/components/hds/flyout/index-test.js (hunk 1 above)
Comment (didoo) [reply showcase/tests/integration/components/hds/flyout/index-test.js:116]: Yes. My suggestion was to make it more explicit in the title of the test. No biggie :)

**Pull request merged**